### PR TITLE
chore(react): Switch to componentDidMount

### PR DESCRIPTION
**Pull request by evanpurkhiser**
Lets see if tests pass

### CODE REVIEW

**Comment by billyvg** [comment]
Ah, looks like the codemod did not change the lifecycle names inside of `createReactClass`.

Can we apply this on getsentry too?

**Comment by EvanPurkhiser** [comment]
Yeah will apply there as well. Just doing this with a quick find replace

**Comment by EvanPurkhiser** [comment]
CWRP will probably be a bit harder

**Comment by billyvg** [comment]
CWRP is riskier, we should not try to bulk update it

**Comment by EvanPurkhiser** [comment]
Yeah I won't

File: src/sentry/static/sentry/app/components/asyncComponent.tsx [hunk 1]
```diff
@@ -115,7 +115,7 @@ export default class AsyncComponent<
     }
   }
 
-  UNSAFE_componentWillMount() {
+  componentDidMount() {
     this.api = new Client();
```
Comment (leedongwei) [inline src/sentry/static/sentry/app/components/asyncComponent.tsx]: Not a blocker, but it's much safer to initiate `api` in the constructor as client is passed down and consumed by its children.

File: src/sentry/static/sentry/app/components/setupWizard.jsx [hunk 2]
```diff
@@ -18,7 +18,7 @@ class SetupWizard extends React.Component {
     this.state = this.getDefaultState();
   }
 
-  UNSAFE_componentWillMount() {
+  componentDidMount() {
     this.api = new Client();
```
Comment (leedongwei) [inline src/sentry/static/sentry/app/components/setupWizard.jsx]: Same as above

File: src/sentry/static/sentry/app/views/organizationDetails/index.jsx [hunk 3]
```diff
@@ -49,7 +49,7 @@ class DeletionPending extends Component {
     this.state = {submitInProgress: false};
   }
 
-  UNSAFE_componentWillMount() {
+  componentDidMount() {
     this.api = new Client();
```
Comment (leedongwei) [inline src/sentry/static/sentry/app/views/organizationDetails/index.jsx]: Same as above

**Review by leedongwei — APPROVED** [review]

File: src/sentry/static/sentry/app/components/bases/pluginComponentBase.jsx [hunk 4]
```diff
@@ -46,7 +46,7 @@ class PluginComponentBase extends React.Component {
     };
   }
 
-  UNSAFE_componentWillMount() {
+  componentDidMount() {
     this.api = new Client();
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/bases/pluginComponentBase.jsx]: This seems like something that should be happening in the constructor.

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/bases/pluginComponentBase.jsx]: Yeah agreed. Or maybe we can just wrap this whole thing with `withApi` and be done with it

Comment (billyvg) [reply src/sentry/static/sentry/app/components/bases/pluginComponentBase.jsx]: Subclasses use the class prop though

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/bases/pluginComponentBase.jsx]: ahh I see, nvm

**Comment by EvanPurkhiser** [comment]
Need to figure out how to fix the builds :/

**Comment by EvanPurkhiser** [comment]
This needs a bit more work. Closing.

**Pull request closed without merging**